### PR TITLE
Create general test methods for publications

### PR DESCRIPTION
**Pull request by index-git**

### CODE REVIEW

File: src/layman/rest_multipublication_test.py [hunk 1]
```diff
@@ -16,62 +11,49 @@ class TestDeletePublicationsClass:
     authn_headers_owner = process_client.get_authz_headers(owner)
     authn_headers_deleter = process_client.get_authz_headers(deleter)
 
-    @pytest.fixture(scope="class")
+    @pytest.fixture()
```
Comment (jirik) [inline src/layman/rest_multipublication_test.py:14]: Why did you remove `scope="class"`?

File: src/layman/rest_publication_test.py [hunk 2]
```diff
@@ -58,9 +53,9 @@ class TestSoapClass:
     access_rights_rewo = {'read': f"{username},EVERYONE", 'write': f"{username}"}
     access_rights_rewe = {'read': f"{username},EVERYONE", 'write': f"{username},EVERYONE"}
 
-    @pytest.fixture(scope='class')
+    @pytest.fixture()
```
Comment (jirik) [inline src/layman/rest_publication_test.py:56]: Why did you remove scope="class"?

File: test/process_client.py [hunk 3]
```diff
@@ -70,66 +76,23 @@ def wait_for_rest(url, max_attempts, sleeping_time, keys_to_check, headers=None)
...
-
-def patch_layer(username,
-                layername,
-                file_paths=None,
-                headers=None,
-                access_rights=None,
-                title=None,
-                assert_status=True,
-                ):
+def patch_publication(publication_type,
```
Comment (jirik) [inline test/process_client.py:79]: I am not sure about order (publication_type, workspace, publication_name). It's probably more natural to use order as it is in REST API URL, so (workspace, publication_type, publication_name). Let's discuss it together.

Comment (index-git) [reply test/process_client.py:79]: The reason for this order is same as in other functions, we want to use with `partial`. For example in src/layman/common/filesystem/util.py::get_publication_dir and many others in the same module.

File: src/layman/rest_multipublication_test.py (hunk 1 above)
Comment (index-git) [reply src/layman/rest_multipublication_test.py]: Reverted. Solved by setting scope="class" for ensure_layman fixture.

File: src/layman/rest_publication_test.py (hunk 2 above)
Comment (index-git) [reply src/layman/rest_publication_test.py]: Reverted. Solved by setting scope="class" for ensure_layman fixture.

**Review by jirik — APPROVED** [review]
Ready to squash
